### ISSUE TITLE
Question on purchase flow

### ISSUE THREAD

**Issue by DigitalMystery** [j3k0/cordova-plugin-purchase#467]
Is there something similar to finish() that you call if the purchase does not go thru?  My use case is something like this...
- Once inside approved do validation on my server
- After my own validation call .finish()

However, what happens if my call to the server fails?  Is there some way to cancel out the transaction?  I tried looking thru the documentation and issues on here, but could not find anything.

**Comment by j3k0** [comment]
You shouldn't cancel the transaction.

If validation shows that transaction: you can finish.
If you have not been able to deliver the purchase: don't finish, app will retry at next launch.

**Comment by DigitalMystery** [comment]
I actually started to test that out yesterday and from what I noticed is that the no matter what whether I call finish or not, "data.transaction.receipt" was not there on the next launch when approved was called (I was currently testing this on Windows yesterday but am using it for all platforms).  Since I no longer had the receipt I could not finish my calls so I just added some logic to store the receipt local where I could get to it if this happens and then finish my profile update the next time approve is called. I will be running all this thru Android and iOS this week as well.  Not sure if that is how it is suppose to work or not, but seems like it will for me.  If it is you can close the ticket.....

**Comment by reMake000** [comment]
@DigitalMystery 
`store._prepareForValidation(product, function () {
   //Now product will have receipt and you can perform server-side validation
  myServerRequest(product); //Perform "product.finish()" after successful server response
}`

**Comment by DigitalMystery** [comment]
@reMake000 

Ok, I have added that, but whenever I purchase a subscription I get the same receipt (Windows & Android - Have not tested the iOS implementation).  So I am never really able to test the above code.  Is there a way to force the plugin to not use the old receipt?  I saw another issue on here where they spoke about iOS, but not the others.  If you are not sure I will add another issue....
